### PR TITLE
[SuperEditor] Fix crash after changing the gesture mode (Resolves #742)

### LINKED ISSUE #742: [super_editor] - Switching from mouse gesture to mobile gesture causes error
Display `SuperEditor` with mouse gestures:

```dart
return SuperEditor(
      editor: _editor,
      composer: _composer,
      documentLayoutKey: _layoutKey,
      inputSource: DocumentInputSource.ime,
      gestureMode: DocumentGestureMode.mouse,
    );
```

Place the caret in the editor.

Switch from mouse gestures to iOS gestures and hot reload:

```dart
return SuperEditor(
      editor: _editor,
      composer: _composer,
      documentLayoutKey: _layoutKey,
      inputSource: DocumentInputSource.ime,
      gestureMode: DocumentGestureMode.iOS,
    );
```

An error widget is displayed and an error is reported to the console:

```
======== Exception caught by widgets library =======================================================
The following assertion was thrown building _FocusMarker:
setState() or markNeedsBuild() called during build.

This Overlay widget cannot be marked as needing to build because the framework is already in the process of building widgets. A widget can be marked as needing to be built during the build phase only if one of its ancestors is currently building. This exception is allowed because the framework builds parent widgets before children, which means a dirty descendant will always be built. Otherwise, the framework might not visit this widget during this build phase.
The widget on which setState() or markNeedsBuild() was called was: Overlay-[LabeledGlobalKey<OverlayState>#52bb7]
  state: OverlayState#54d1d(entries: [OverlayEntry#fdaed(opaque: true; maintainState: false), OverlayEntry#80f78(opaque: false; maintainState: true), OverlayEntry#1d73d(opaque: false; maintainState: false)])
The widget which was currently being built when the offending call was made was: _FocusMarker
The relevant error-causing widget was: 
  SuperEditor SuperEditor:file:///Users/admin/Projects/jangle/FlutterOS/typewriter/lib/main.dart:89:12
When the exception was thrown, this was the stack: 
#0      Element.markNeedsBuild.<anonymous closure> (package:flutter/src/widgets/framework.dart:4476:11)
#1      Element.markNeedsBuild (package:flutter/src/widgets/framework.dart:4491:6)
#2      State.setState (package:flutter/src/widgets/framework.dart:1129:15)
#3      OverlayState.insert (package:flutter/src/widgets/overlay.dart:359:5)
#4      _IOSDocumentTouchInteractorState._showEditingControlsOverlay (package:super_editor/src/default_editor/document_gestures_touch_ios.dart:817:26)
#5      _IOSDocumentTouchInteractorState.initState (package:super_editor/src/default_editor/document_gestures_touch_ios.dart:144:7)
#6      StatefulElement._firstBuild (package:flutter/src/widgets/framework.dart:4942:57)
#7      ComponentElement.mount (package:flutter/src/widgets/framework.dart:4781:5)
#8      Element.inflateWidget (package:flutter/src/widgets/framework.dart:3817:16)
#9      Element.updateChild (package:flutter/src/widgets/framework.dart:3545:20)
#10     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:6222:14)
#11     Element.updateChild (package:flutter/src/widgets/framework.dart:3530:15)
#12     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4832:16)
#13     Element.rebuild (package:flutter/src/widgets/framework.dart:4529:5)
#14     ProxyElement.update (package:flutter/src/widgets/framework.dart:5154:5)
#15     _InheritedNotifierElement.update (package:flutter/src/widgets/inherited_notifier.dart:108:11)
#16     Element.updateChild (package:flutter/src/widgets/framework.dart:3530:15)
#17     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4832:16)
#18     StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:4977:11)
#19     Element.rebuild (package:flutter/src/widgets/framework.dart:4529:5)
#20     StatefulElement.update (package:flutter/src/widgets/framework.dart:5009:5)
#21     Element.updateChild (package:flutter/src/widgets/framework.dart:3530:15)
#22     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4832:16)
#23     StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:4977:11)
#24     Element.rebuild (package:flutter/src/widgets/framework.dart:4529:5)
#25     StatefulElement.update (package:flutter/src/widgets/framework.dart:5009:5)
#26     Element.updateChild (package:flutter/src/widgets/framework.dart:3530:15)
#27     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4832:16)
#28     StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:4977:11)
#29     Element.rebuild (package:flutter/src/widgets/framework.dart:4529:5)
#30     StatefulElement.update (package:flutter/src/widgets/framework.dart:5009:5)
#31     Element.updateChild (package:flutter/src/widgets/framework.dart:3530:15)
#32     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4832:16)
#33     Element.rebuild (package:flutter/src/widgets/framework.dart:4529:5)
#34     ProxyElement.update (package:flutter/src/widgets/framework.dart:5154:5)
#35     Element.updateChild (package:flutter/src/widgets/framework.dart:3530:15)
#36     RenderObjectElement.updateChildren (package:flutter/src/widgets/framework.dart:5825:32)
#37     MultiChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:6375:17)
#38     Element.updateChild (package:flutter/src/widgets/framework.dart:3530:15)
#39     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4832:16)
#40     Element.rebuild (package:flutter/src/widgets/framework.dart:4529:5)
#41     ProxyElement.update (package:flutter/src/widgets/framework.dart:5154:5)
#42     Element.updateChild (package:flutter/src/widgets/framework.dart:3530:15)
#43     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:6222:14)
#44     Element.updateChild (package:flutter/src/widgets/framework.dart:3530:15)
#45     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4832:16)
#46     Element.rebuild (package:flutter/src/widgets/framework.dart:4529:5)
#47     StatelessElement.update (package:flutter/src/widgets/framework.dart:4883:5)
#48     Element.updateChild (package:flutter/src/widgets/framework.dart:3530:15)
#49     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4832:16)
#50     Element.rebuild (package:flutter/src/widgets/framework.dart:4529:5)
#51     StatelessElement.update (package:flutter/src/widgets/framework.dart:4883:5)
#52     Element.updateChild (package:flutter/src/widgets/framework.dart:3530:15)
#53     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4832:16)
#54     Element.rebuild (package:flutter/src/widgets/framework.dart:4529:5)
#55     ProxyElement.update (package:flutter/src/widgets/framework.dart:5154:5)
#56     Element.updateChild (package:flutter/src/widgets/framework.dart:3530:15)
#57     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4832:16)
#58     Element.rebuild (package:flutter/src/widgets/framework.dart:4529:5)
#59     ProxyElement.update (package:flutter/src/widgets/framework.dart:5154:5)
#60     Element.updateChild (package:flutter/src/widgets/framework.dart:3530:15)
#61     RenderObjectElement.updateChildren (package:flutter/src/widgets/framework.dart:5825:32)
#62     MultiChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:6375:17)
#63     Element.updateChild (package:flutter/src/widgets/framework.dart:3530:15)
#64     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4832:16)
#65     StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:4977:11)
#66     Element.rebuild (package:flutter/src/widgets/framework.dart:4529:5)
#67     StatefulElement.update (package:flutter/src/widgets/framework.dart:5009:5)
#68     Element.updateChild (package:flutter/src/widgets/framework.dart:3530:15)
#69     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4832:16)
#70     Element.rebuild (package:flutter/src/widgets/framework.dart:4529:5)
#71     ProxyElement.update (package:flutter/src/widgets/framework.dart:5154:5)
#72     Element.updateChild (package:flutter/src/widgets/framework.dart:3530:15)
#73     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4832:16)
#74     StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:4977:11)
#75     Element.rebuild (package:flutter/src/widgets/framework.dart:4529:5)
#76     StatefulElement.update (package:flutter/src/widgets/framework.dart:5009:5)
#77     Element.updateChild (package:flutter/src/widgets/framework.dart:3530:15)
#78     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:6222:14)
#79     Element.updateChild (package:flutter/src/widgets/framework.dart:3530:15)
#80     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4832:16)
#81     Element.rebuild (package:flutter/src/widgets/framework.dart:4529:5)
#82     ProxyElement.update (package:flutter/src/widgets/framework.dart:5154:5)
#83     Element.updateChild (package:flutter/src/widgets/framework.dart:3530:15)
#84     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:6222:14)
#85     Element.updateChild (package:flutter/src/widgets/framework.dart:3530:15)
#86     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4832:16)
#87     StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:4977:11)
#88     Element.rebuild (package:flutter/src/widgets/framework.dart:4529:5)
#89     StatefulElement.update (package:flutter/src/widgets/framework.dart:5009:5)
#90     Element.updateChild (package:flutter/src/widgets/framework.dart:3530:15)
#91     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4832:16)
#92     StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:4977:11)
#93     Element.rebuild (package:flutter/src/widgets/framework.dart:4529:5)
#94     StatefulElement.update (package:flutter/src/widgets/framework.dart:5009:5)
#95     Element.updateChild (package:flutter/src/widgets/framework.dart:3530:15)
#96     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4832:16)
#97     Element.rebuild (package:flutter/src/widgets/framework.dart:4529:5)
#98     ProxyElement.update (package:flutter/src/widgets/framework.dart:5154:5)
#99     Element.updateChild (package:flutter/src/widgets/framework.dart:3530:15)
#100    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4832:16)
#101    Element.rebuild (package:flutter/src/widgets/framework.dart:4529:5)
#102    ProxyElement.update (package:flutter/src/widgets/framework.dart:5154:5)
#103    Element.updateChild (package:flutter/src/widgets/framework.dart:3530:15)
#104    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4832:16)
#105    StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:4977:11)
#106    Element.rebuild (package:flutter/src/widgets/framework.dart:4529:5)
#107    StatefulElement.update (package:flutter/src/widgets/framework.dart:5009:5)
#108    Element.updateChild (package:flutter/src/widgets/framework.dart:3530:15)
#109    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4832:16)
#110    Element.rebuild (package:flutter/src/widgets/framework.dart:4529:5)
#111    ProxyElement.update (package:flutter/src/widgets/framework.dart:5154:5)
#112    Element.updateChild (package:flutter/src/widgets/framework.dart:3530:15)
#113    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4832:16)
#114    StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:4977:11)
#115    Element.rebuild (package:flutter/src/widgets/framework.dart:4529:5)
#116    StatefulElement.update (package:flutter/src/widgets/framework.dart:5009:5)
#117    Element.updateChild (package:flutter/src/widgets/framework.dart:3530:15)
#118    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4832:16)
#119    StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:4977:11)
#120    Element.rebuild (package:flutter/src/widgets/framework.dart:4529:5)
#121    BuildOwner.buildScope (package:flutter/src/widgets/framework.dart:2659:19)
#122    WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:891:21)
#123    RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:370:5)
#124    SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1146:15)
#125    SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1083:9)
#126    SchedulerBinding.scheduleWarmUpFrame.<anonymous closure> (package:flutter/src/scheduler/binding.dart:864:7)
(elided 4 frames from class _RawReceivePortImpl, class _Timer, and dart:async-patch)
====================================================================================================
```

### PR DESCRIPTION
**Pull request by angelosilvestre**
[SuperEditor] Fix crash after changing the gesture mode. Resolves #742

When `SuperEditor` has focus, if we change the gesture mode and hot reload it crashes the app.

The crash is happening because, when we change the gesture mode and hot reload, the `initState` of the new gesture interactor is called while the app `Overlay` is being built, so we can't show the editing controls overlay in the same frame.

This PR changes the mobile interactors to show the editing controls overlay in a post frame callback.

Also, the iOS interactor is adding a listener to the scroll position during the `build` method. In the first build of the interactor after the gesture mode change, the `ScrollController` seems to be also attached to the old interactor, which also crashes the app.  I'm not sure why we are doing that, as in the android interactor we aren't doing that.

https://github.com/superlistapp/super_editor/blob/d5e7460956fc6d60abd65991d48ca2c073c1da38/super_editor/lib/src/default_editor/document_gestures_touch_ios.dart#L1087-L1094

This PR changes the build to defer adding the listener to the next frame.

I'm not sure how to write a test for this.

### CODE REVIEW

File: super_editor/lib/src/default_editor/document_gestures_touch_ios.dart [hunk 1]
```diff
@@ -1086,9 +1097,18 @@ class _IOSDocumentTouchInteractorState extends State<IOSDocumentTouchInteractor>
...
-      if (scrollPosition != _activeScrollPosition) {
-        _activeScrollPosition = scrollPosition;
-        _activeScrollPosition?.addListener(_onScrollChange);
+      if (_scrollController.positions.length > 1) {
+        // During Hot Reload, if the gesture mode was changed,
+        // the widget might be built while the old gesture interactor
+        // scroller is still attached to the _scrollController.
+        //
+        // Defer adding the listener to the next frame.
+        setState(() {});
```
Comment (matthew-carroll) [inline super_editor/lib/src/default_editor/document_gestures_touch_ios.dart:1106]: Were you able to run this code path? I think calling `setState()` from `build()` will crash.

Comment (angelosilvestre) [reply super_editor/lib/src/default_editor/document_gestures_touch_ios.dart:1106]: At least in my manual tests this code runs without errors. I could change it to call  `setState()` in a post frame callback.

Comment (matthew-carroll) [reply super_editor/lib/src/default_editor/document_gestures_touch_ios.dart:1106]: Yeah, let's add a post frame callback.

Comment (angelosilvestre) [reply super_editor/lib/src/default_editor/document_gestures_touch_ios.dart]: Updated.

**Review by matthew-carroll — APPROVED** [review]
LGTM

**Pull request merged**